src/FactoryTrait.php

```diff
@@ -115,32 +119,33 @@ function ($o) {
      */
     public function normalizeClassName($name, $prefix = null)
     {
-        if (is_array($name) && isset($name[0])) {
-            $name[0] = $this->normalizeClassName($name[0]);
+        if (!$name) {
+            if (
+                isset($this->_appScopeTrait, $this->app)
+                && method_exists($this->app, 'normalizeClassNameApp')
+            ) {
+                $name = $this->app->normalizeClassNameApp($name);
+            }
 
             return $name;
         }
 
+        if ($name[0] != '/' && strpos($name, '\\') === false && $prefix) {
+            // Add prefix
+
+            $name = $prefix.'/'.$name;
+        }
+
         if (
-            is_string($name)
+            $name[0] != '/'
+            && strpos($name, '\\') === false
             && isset($this->_appScopeTrait, $this->app)
             && method_exists($this->app, 'normalizeClassNameApp')
         ) {
-            $name = $this->app->normalizeClassNameApp($name, $prefix);
-        }
-
-        if (!is_string($name)) {
-            return $name;
+            $name = $this->app->normalizeClassNameApp($name);
         }
 
         $name = str_replace('/', '\\', $name);
```

Comment:
1. I think code can be optimized in this method by merging 1st and last blocks together.
2. Also this str_replace line will currently not be executed in case $name argument is not set in this method (see above). That's bad for consistency I guess.

Reply:
not sure if this is addressed or not.